aria_data_loaders/aria_data_utils/aria_streamer.py

```diff
@@ -0,0 +1,842 @@
+
+            if should_rec_timestamp_from_user:
+                val = input("o to skip, c to record")
+                if val == "c":
+                    self.vrs_idx_of_interest_list.append(frame_idx)
+
+        return img_list, img_metadata_list, cpf_T_marker_list
+
+    def initialize_trajectory(self):
+        # frame(ariaWorld) is same as frame(device) at the start
+        cpf_T_ariaWorld = self.cpf_T_device
+
+        for i in range(len(self.mps_trajectory)):
+            self.trajectory_s[i] = self.mps_trajectory[
+                i
+            ].tracking_timestamp.total_seconds()
+            ariaWorld_T_device = sp.SE3(
+                self.mps_trajectory[i].transform_world_device.to_matrix()
+            )
+            self.ariaWorld_T_device_trajectory.append(ariaWorld_T_device)
+
+            ariaWorld_T_cpf = ariaWorld_T_device * self.device_T_cpf
+
+            ariaCorrectedWorld_T_device = cpf_T_ariaWorld * ariaWorld_T_device
+            self.ariaCorrectedWorld_T_device_trajectory.append(
+                ariaCorrectedWorld_T_device
+            )
+
+            ariaCorrectedWorld_T_cpf = ariaCorrectedWorld_T_device * self.device_T_cpf
+            self.ariaCorrectedWorld_T_cpf_trajectory.append(ariaCorrectedWorld_T_cpf)
+
+            self.xyz_trajectory[i, :] = ariaWorld_T_cpf.translation()
+            # self.quat_trajectory[i,:] = self.mps_trajectory[i].transform_world_device.quaternion()
+        assert len(self.trajectory_s) == len(
+            self.ariaCorrectedWorld_T_device_trajectory
+        )
+
+    # TODO: Can be optimized. This is making O(n^2)
+    def get_closest_mps_idx_to_timestamp_ns(self, timestamp_ns_of_interest: int):
+        mps_idx_of_interest = np.argmin(
+            np.abs(self.trajectory_s * 1e9 - timestamp_ns_of_interest)
+        )
+        return mps_idx_of_interest
+
+    def get_closest_world_T_device_to_timestamp(self, timestamp_ns_of_interest: int):
+        """
+        timestamp_of_interest -> nanoseconds
+        """
+        mps_idx_of_interest = self.get_closest_mps_idx_to_timestamp_ns(
+            timestamp_ns_of_interest
+        )
+        sp_transform_of_interest = self.ariaWorld_T_device_trajectory[
+            mps_idx_of_interest
+        ]
+        return sp_transform_of_interest
+
+    def get_closest_ariaCorrectedWorld_T_cpf_to_timestamp(
+        self, timestamp_ns_of_interest: int
+    ):
+        """
+        timestamp_of_interest -> nanoseconds
+        """
+        mps_idx_of_interest = self.get_closest_mps_idx_to_timestamp_ns(
+            timestamp_ns_of_interest
+        )
+        sp_transform_of_interest = self.ariaCorrectedWorld_T_cpf_trajectory[
+            mps_idx_of_interest
+        ]
+        return sp_transform_of_interest
+
+    def get_avg_ariaCorrectedWorld_T_marker(
+        self,
+        img_list: List,
+        img_metadata_list: List,
+        cpf_T_marker_list: List,
+        filter_dist: float = 2.4,
```

Comment:
Should we make this a param set through config later?

Reply:
Thanks for this one, not sure of config is the best place for this so making a global constant (for the file).

@zephirefaith @jimmytyyang  what do you think? Is this important enough to go in config?

Reply:
@KavitShah1998 Is this resolved?